documentation/glossary.md

```diff
@@ -0,0 +1,300 @@
+# Glossary
+
+Definitions of terms used in Bolt documentation.
+
+## agent
+
+The Puppet agent software package, also known as `puppet-agent`.
+
+## apply
+
+`Apply` is used to apply a block of manifest code to the targets.
+
+Applying manifest code requires facts to compile a catalog. Targets must also
+have the Puppet agent package installed to apply manifest code. To prep targets
+for an apply, call the `apply_prep` function before the apply function.
+
+📖 **Related information**
+
+- [`apply` example](https://puppet.com/docs/bolt/latest/plan_functions.html#apply)
+
+## apply_prep
+
+Installs the `puppet-agent` package on targets if needed, then collects facts,
+including any custom facts found in Bolt's module path.
+
+📖 **Related information**
+
+- [`apply_prep` example](https://puppet.com/docs/bolt/latest/plan_functions.html#apply-prep)
+
+## catalog
+
+A catalog is a file that describes the desired state of each managed resource on
+a target. It is a compilation of all the resources that the Puppet agent applies
+to a given target, as well as the relationships between those resources.
+
+Catalogs are compiled by Bolt from manifests and agent-provided data during an
+apply. Bolt then serves the compiled catalog to the agent on a target.
+
+Unlike the manifests from which they are compiled, catalogs don't contain any
+conditional logic or functions. They are unambiguous, relevant to only a
+specific target, and generated by a target.
+
+📖 **Related information**
+
+- [Catalog compilation](https://puppet.com/docs/puppet/latest/subsystem_catalog_compilation.html#subsystem_catalog_compilation)
+
+## controller
```

Comment:
I believe this is the only term I couldn't find a definition for

Reply:
It's the 'thing' (computer, VM, container, etc.) running Bolt. I'm not sure on an exact definition here, but @hestonhoffman might have some suggestions.

Reply:
How about:
> The device you're using to run Bolt. This could be a laptop, a virtual machine, a container,
or even a network device or Raspberry Pi!